sky/spot/recovery_strategy.py

```diff
@@ -444,3 +450,77 @@ def recover(self) -> float:
                         f'{self._MAX_RETRY_CNT} times.')
 
             return job_submitted_at
+
+
+class EagerFailoverStrategyExecutor(FailoverStrategyExecutor,
+                                    name='EAGER_FAILOVER',
+                                    default=True):
+    """Aggressive failover strategy
+
+    This strategy is an extension of the failover strategy. Instead of waiting
+    in the same region when the preemption happens, it immediately terminates
+    the cluster and relaunches it in a different regions. This is based on the
+    observation that the preemption is likely to happen again shortly in the
+    same region, so trying other regions first is more likely to get a longer
+    running cluster.
+    """
+
+    def recover(self) -> float:
+        # 1. Terminate the current cluster
+        # 2. Launch the cluster without retrying the previously launched region
+        # 3. Launch the cluster with no cloud/region constraint or respect the
+        #    original user specification.
+
+        # Step 1
+        logger.debug('Terminating unhealthy spot cluster and '
+                     'reset cloud region.')
+        terminate_cluster(self.cluster_name)
+
+        # Step 2
+        logger.debug('Relaunch the cluster skipping the previously launched '
+                     'cloud/region.')
+        if self._launched_cloud_region is not None:
+            task = self.dag.tasks[0]
+            resources = list(task.resources)[0]
+            if resources.region is None and resources.zone is None:
+                # Optimization: We only block the previously launched region,
+                # if the task does not specify a region or zone, because,
+                # otherwise, we will spend unnecessary time for skipping the
+                # only specified region/zone.
+                launched_cloud, launched_region = self._launched_cloud_region
+                task.blocked_resources = {
+                    resources.copy(cloud=launched_cloud, region=launched_region)
```

Comment:
Is it intentional that this policy does not retry in other zones of the same region, i.e., it is an eager region jumping policy? Worth a comment.

Reply:
Actually, imagine an important user that explicitly sets `region: xxx`. If a zone is particularly preemptive, then the current impl is likely to go to that zone again (since it was chosen initially, let's assume it's cheapest/alphabetically smallest).

Should we handle this case too? The best thing we came up with before was go to the next region but a randomized zone.

That said we can leave this to a future PR. This new strategy behaves the same as the previous strategy in this hypothetical case.

Reply:
Sounds good.